plugins/modules/helm_pull.py

```diff
@@ -0,0 +1,335 @@
+author:
+  - Aubin Bikouo (@abikouo)
+description:
+  - Retrieve a package from a package repository, and download it locally.
+  - It can also be used to perform cryptographic verification of a chart without installing the chart.
+  - There are options for unpacking the chart after download.
+
+requirements:
+  - "helm >= 3.0 (https://github.com/helm/helm/releases)"
+
+options:
+  chart_ref:
+    description:
+      - chart name on chart repository.
+      - absolute URL.
+    required: true
+    type: str
+  chart_version:
+    description:
+      - Specify a version constraint for the chart version to use.
+      - This constraint can be a specific tag (e.g. 1.1.1) or it may reference a valid range (e.g. ^2.0.0).
+      - Mutually exclusive with C(chart_devel).
+    type: str
+  verify_chart:
+    description:
+      - Verify the package before using it.
+    default: False
+    type: bool
+  verify_chart_keyring:
+    description:
+      - location of public keys used for verification.
+    type: path
+  provenance:
+    description:
+      - Fetch the provenance file, but don't perform verification.
+    type: bool
+    default: False
+  repo_url:
+    description:
+      - chart repository url where to locate the requested chart.
+    type: str
+    aliases: [ url, chart_repo_url ]
+  repo_username:
+    description:
+      - Chart repository username where to locate the requested chart.
+      - Required if C(repo_password) is specified.
+    type: str
+    aliases: [ username, chart_repo_username ]
+  repo_password:
+    description:
+      - Chart repository password where to locate the requested chart.
+      - Required if C(repo_username) is specified.
+    type: str
+    aliases: [ password, chart_repo_password ]
+  pass_credentials:
+    description:
+      - Pass credentials to all domains.
+    default: False
+    type: bool
+  skip_tls_certs_check:
+    description:
+    - Whether or not to check tls certificate for the chart download.
+    - Requires helm >= 3.3.0.
+    type: bool
+    default: False
+  chart_devel:
+    description:
+    - Use development versions, too. Equivalent to version '>0.0.0-0'.
+    - Mutually exclusive with C(chart_version).
+    type: bool
+  untar_chart:
+    description:
+    - if set to true, will untar the chart after downloading it.
+    type: bool
+    default: False
+  destination:
+    description:
+    - location to write the chart.
+    type: path
+    required: True
+  chart_ca_cert:
+    description:
+    - Verify certificates of HTTPS-enabled servers using this CA bundle.
+    - Requires helm >= 3.1.0.
+    type: path
+  chart_ssl_cert_file:
+    description:
+    - Identify HTTPS client using this SSL certificate file.
+    - Requires helm >= 3.1.0.
+    type: path
+  chart_ssl_key_file:
+    description:
+    - Identify HTTPS client using this SSL key file
+    - Requires helm >= 3.1.0.
+    type: path
+
+extends_documentation_fragment:
+  - kubernetes.core.helm_common_options
```

Comment:
Does this actually use any of these options? Unless I'm missing something, it doesn't seem like it's necessary.

Reply:
Removed.